azurerm/internal/services/network/resource_arm_virtual_hub_connection.go

```diff
@@ -0,0 +1,301 @@
+	if newConnection == nil {
+		return fmt.Errorf("Connection %q was not found in Virtual Hub %q / Resource Group %q", name, id.Name, resourceGroup)
+	}
+	if newConnection.ID == nil && *newConnection.ID == "" {
+		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `id` was nil or empty", name, id.Name, resourceGroup)
+	}
+
+	d.SetId(*newConnection.ID)
+	return resourceArmVirtualHubConnectionRead(d, meta)
+}
+
+func resourceArmVirtualHubConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VirtualHubClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ParseVirtualHubConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	virtualHubName := id.VirtualHubName
+
+	virtualHub, err := client.Get(ctx, resourceGroup, virtualHubName)
+	if err != nil {
+		if utils.ResponseWasNotFound(virtualHub.Response) {
+			log.Printf("[DEBUG] Virtual Hub %q was not found in Resource Group %q - so Connection %q can't exist - removing from state", id.Name, virtualHubName, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", virtualHubName, resourceGroup, err)
+	}
+
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", virtualHubName, resourceGroup)
+	}
+
+	if virtualHub.VirtualHubProperties.VirtualNetworkConnections == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties.VirtualNetworkConnections` was nil", virtualHubName, resourceGroup)
+	}
+	connections := *virtualHub.VirtualHubProperties.VirtualNetworkConnections
+
+	connection, _ := findVirtualHubConnection(id.Name, connections)
+	if connection == nil {
+		log.Printf("[DEBUG] Connection %q was not found within Virtual Hub %q (Resource Group %q) - removing from state", id.Name, virtualHubName, resourceGroup)
+		d.SetId("")
+		return nil
+	}
+	if connection.HubVirtualNetworkConnectionProperties == nil {
+		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `properties` was nil`", id.Name, virtualHubName, resourceGroup)
+	}
+	props := *connection.HubVirtualNetworkConnectionProperties
+
+	d.Set("name", id.Name)
+	d.Set("virtual_hub_id", virtualHub.ID)
+	d.Set("hub_to_vitual_network_traffic_allowed", props.AllowHubToRemoteVnetTransit)
+	d.Set("vitual_network_to_hub_gateways_traffic_allowed", props.AllowRemoteVnetToUseHubVnetGateways)
+	d.Set("internet_security_enabled", props.EnableInternetSecurity)
+
+	remoteVirtualNetworkId := ""
+	if props.RemoteVirtualNetwork != nil && props.RemoteVirtualNetwork.ID != nil {
+		remoteVirtualNetworkId = *props.RemoteVirtualNetwork.ID
+	}
+	d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
```

Comment:
Could we be consistent with other resources
```suggestion
	if connection.HubVirtualNetworkConnectionProperties == nil {
		return fmt.Errorf("Error retrieving Connection %q (Virtual Hub %q / Resource Group %q): `properties` was nil`", id.Name, virtualHubName, resourceGroup)
	}

	d.Set("name", id.Name)
	d.Set("virtual_hub_id", virtualHub.ID)
		
	if props := connection.HubVirtualNetworkConnectionProperties; props != nil {
		d.Set("hub_to_vitual_network_traffic_allowed", props.AllowHubToRemoteVnetTransit)
		d.Set("vitual_network_to_hub_gateways_traffic_allowed", props.AllowRemoteVnetToUseHubVnetGateways)
		d.Set("internet_security_enabled", props.EnableInternetSecurity)

		remoteVirtualNetworkId := ""
		if props.RemoteVirtualNetwork != nil && props.RemoteVirtualNetwork.ID != nil {
			remoteVirtualNetworkId = *props.RemoteVirtualNetwork.ID
		}
		d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
	}
```

Reply:
Updated